server/src/main/java/org/apache/druid/server/http/CoordinatorCompactionConfigsResource.java

```diff
@@ -119,6 +122,11 @@ public Response addOrUpdateCompactionConfig(
           .getCompactionConfigs()
           .stream()
           .collect(Collectors.toMap(DataSourceCompactionConfig::getDataSource, Function.identity()));
+      CompactionConfigValidationResult validationResult =
+          ClientCompactionRunnerInfo.validateCompactionConfig(newConfig, current.getEngine());
+      if (!validationResult.isValid()) {
+        throw InvalidInput.exception("Compaction config not supported. Reason[%s].", validationResult.getReason());
```

Comment:
Just throwing an exception here takes away control over the HTTP status code.
Do this instead:

```
return ServletResourceUtils.buildErrorResponseFrom(InvalidInput.exception(...));
```